components/search/src/types.rs

```diff
@@ -0,0 +1,197 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// The list of possible application names that are currently supported.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchApplicationName {
+    Firefox = 1,
+    FirefoxAndroid = 2,
+    FocusAndroid = 3,
+    FirefoxIOS = 4,
+    FocusIOS = 5,
+}
+
+impl SearchApplicationName {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            SearchApplicationName::Firefox => "firefox",
+            SearchApplicationName::FirefoxAndroid => "firefox-android",
+            SearchApplicationName::FocusAndroid => "focus-android",
+            SearchApplicationName::FirefoxIOS => "firefox-ios",
+            SearchApplicationName::FocusIOS => "focus-ios",
+        }
+    }
+}
+
+/// The list of possible update channels for a user's build.
+/// Use `default` for a self-build or an unknown channel.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchUpdateChannel {
+    Default,
+    Nightly,
+    Aurora,
+    Beta,
+    Release,
+    ESR,
```

Comment:
```suggestion
    Default = 1,
    Nightly = 2,
    Aurora = 3,
    Beta = 4,
    Release = 5,
    ESR = 6,
```

Let's give these variants fixed discriminants, too—unless you wanted to specifically avoid them for `SearchUpdateChannel` but not `SearchApplicationName`?